src/plugins/WebOfTrust/util/DelayedBackgroundJob.java

```diff
@@ -0,0 +1,270 @@
+        if (wait < 0) {
+            wait = 0;
+        }
+        ticker.queueTimedJob(new FastRunnable() {
+            @Override
+            public void run() {
+                executor.execute(realJob, name + " (running)");
+            }
+        }, name + " (waiting)", wait, true, false);
+        toWAITING();
+    }
+
+    @Override
+    public synchronized void terminate() {
+        switch (state) {
+            case TERMINATED:
+                assert(thread == null) : "having job thread in TERMINATED state";
+                return;
+            case TERMINATING:
+                assert(thread != null) : "TERMINATING state but no thread";
+                return;
+            case IDLE:
+                toTERMINATED();
+                return;
+            case WAITING:
+                toTERMINATED();
+                break;
+            case RUNNING:
+                toTERMINATING();
+                break;
+        }
+    }
+
+    @Override
+    public synchronized boolean isTerminated() {
+        return state == JobState.TERMINATED;
+    }
+
+    @Override
+    public synchronized void waitForTermination(long timeout) throws InterruptedException {
+        long deadline = System.currentTimeMillis() + timeout;
+        while(timeout >= 0 || state != JobState.TERMINATED) {
+            wait(timeout);
+            timeout = deadline - System.currentTimeMillis();
+        }
+    }
+
+    /**
+     * {RUNNING} -> IDLE
+     */
+    private void toIDLE() {
+        assert(state == JobState.RUNNING) : "going to IDLE from non-RUNNING state";
+        assert(thread == Thread.currentThread()) : "going to IDLE from non-job thread";
+        thread = null;
+        state = JobState.IDLE;
+    }
+
+    /**
+     * {IDLE} -> WAITING
+     */
+    private void toWAITING() {
+        assert(state == JobState.IDLE) : "going to WAITING from non-IDLE state";
+        assert(thread == null) : "having job thread while going to WAITING state";
+        state = JobState.WAITING;
+    }
+
+    /**
+     * {WAITING} -> RUNNING
+     */
+    private void toRUNNING() {
+        assert(state == JobState.WAITING) : "going to RUNNING state from non-WAITING state";
+        assert(thread == null) : "already having job thread while going to RUNNING state";
+        thread = Thread.currentThread();
+        state = JobState.RUNNING;
+    }
+
+    /**
+     * {RUNNING} -> TERMINATING
+     */
+    private void toTERMINATING() {
+        assert(state == JobState.RUNNING);
+        assert(thread != null) : "going to TERMINATING state without job thread";
+        thread.interrupt();
+    }
+
+    /**
+     * {IDLE, TERMINATING, WAITING} -> TERMINATED
+     */
+    private void toTERMINATED() {
+        if (state == JobState.TERMINATING) {
+            assert(thread == Thread.currentThread()) : "TERMINATING from non-job thread";
+            thread = null;
+        } else {
+            assert (state == JobState.IDLE || state == JobState.WAITING)
```

Comment:
Didn't look at the particular commit, but the full review of the class which I did later than the commit showed that this is not an issue anymore.